src/tests/system/tests/test_sssctl.py

```diff
@@ -241,38 +154,16 @@ def test_sssctl__check_missing_domain_name(client: Client):
     assert ex.match(r"Section \[domain\/\] is not allowed. Check for typos.*"), "Wrong error message was returned"
 
 
-@pytest.mark.importance("high")
 @pytest.mark.tools
 @pytest.mark.topology(KnownTopology.Client)
-def test_sssctl__check_misplaced_option(client: Client):
-    """
-    :title: sssctl config-check detects misplaced option
-    :setup:
-        1. In domain set "services" to "nss, pam"
-        2. Start SSSD, without config check
-    :steps:
-        1. Call sssctl config-check
-        2. Check error message
-    :expectedresults:
-        1. config-check detects an error in config
-        2. Error message is properly set
-    :customerscenario: False
-    """
-    client.sssd.common.local()
-    client.sssd.dom("test")["services"] = "nss, pam"
-
-    client.sssd.start(check_config=False)
-
-    result = client.sssctl.config_check()
-    assert result.rc != 0, "Config-check did not detect misconfigured config"
-
-    pattern = re.compile(r".Attribute 'services' is not allowed in section .*")
-    assert pattern.search(result.stdout), "Wrong error message was returned"
-
-
-@pytest.mark.tools
-@pytest.mark.topology(KnownTopology.Client)
-def test_sssctl__check_invalid_option_name_in_snippet(client: Client):
+@pytest.mark.parametrize(
+    "contents,expected",
+    [
+        ("[domain/local]\ninvalid_option = True", "Attribute 'invalid_option' is not allowed"),
+        ("[invalid/local]\ninvalid_option = True", "Section [invalid/local] is not allowed"),
+    ],
+)
+def test_sssctl__check_invalid_option_name_in_snippet(client: Client, contents, expected):
     """
     :title: sssctl config-check detects invalid option name in snippet
```

Comment:
"""
    :title: sssctl config-check validates configuration snippet
    :setup:
        1. Create a config snippet with an invalid option
    :steps:
        1. Check the configuration using sssctl
    :expectedresults:
        1. The config check fails with the appropriate output
    :customerscenario: True
    """